pkg/gardenlet/controller/shoot/shoot/reconciler_migrate.go

```diff
@@ -271,7 +268,7 @@ func (r *Reconciler) runMigrateShootFlow(ctx context.Context, o *operation.Opera
 		})
 		deleteKubeAPIServer = g.Add(flow.Task{
 			Name:         "Deleting kube-apiserver deployment",
-			Fn:           flow.TaskFn(botanist.DeleteKubeAPIServer).SkipIf(!kubeAPIServerDeploymentFound),
+			Fn:           flow.TaskFn(botanist.DeleteKubeAPIServer),
```

Comment:
In `reconciler_delete.go` we do this:
```suggestion
			Fn:           flow.TaskFn(botanist.DeleteKubeAPIServer).RetryUntilTimeout(defaultInterval, defaultTimeout),
```
does it make sense here too?

Reply:
Generally, the migrate flow is a lot shorter than the delete so not having a `RetryUntilTimeout()` isn't a big deal, but I can add it here for consistency.